Gopkg.toml

```diff
@@ -296,3 +296,7 @@
 [[constraint]]
   branch = "master"
   name = "github.com/cisco-ie/nx-telemetry-proto"
+
+[[constraint]]
+  name = "github.com/signalfx/golib"
+  version = "2.0.0"
```

Comment:
Can you set this to `branch = "master"` with a comment that we are waiting on 2.3.5 or later for dependency changes.  I'll make a note to update this before the next release.

Reply:
Done. PTAL

Reply:
```
# Gopkg.lock is out of sync:
github.com/signalfx/golib@v2.0.0: not allowed by constraint master
```

Reply:
Hmmm, not sure what is going on here (because I'm not very familiar with `dep`). It  looks like even though I updated `Gopkg.toml` to specify `branch = "master"`, and then ran `make deps`, the `Gopkg.lock` still has this:
```
[[projects]]
  digest = "1:3d52a2ff8fabf5cb15a88e8451a524af87f203c80ad8a33209fd52e15eaa4997"
  name = "github.com/signalfx/golib"
  packages = [
    "datapoint",
    "errors",
    "event",
    "eventcounter",
    "log",
    "sfxclient",
    "timekeeper",
    "trace",
    "trace/format",
  ]
  pruneopts = ""
  revision = "271e7b77c46c2cb201be901a302cb181a9d1645a"
  version = "v2.0.0"
```

I suspect the version shouldn't be here any more after a successful run of `dep ensure...`

@danielnelson - any pointers on what to do next?

Reply:
I get a couple errors running `dep ensure`, first this one:
```
master: Could not introduce github.com/signalfx/golib@master, as it has a dependency on github.com/go-logfmt/logfmt with constraint ^0.3.0, which has no overlap with existing constraint ^0.4.0 from (root)
```

This can be fixed by changing Telegraf's logfmt dependency to `">=0.3.0, <0.5.0"`,  or another fix would be to do the same on signalfx.  It doesn't appear to cause an issue, tests are still passing.

After changing this I get:
```
master: Could not introduce github.com/signalfx/golib@master, as it has a dependency on github.com/golang/protobuf with constraint 1643683e1b54a9e88ad26d98f81400c8c9d9f4f9, which has no overlap with existing constraint ^1.1.0 from (root)
```

This one probably needs to be fixed in signalfx: `version = "1.0.0" or such, since they are saying only this revision of protobuf will work, which is too strict and I expect any version >= 1.0.0 (the first release) would be fine.  At any rate I believe we need 1.1.0 due to an existing dependency.  It looks like this revision is a bit older than 1.0.0, but maybe they would be willing to move it forward so that we can pin a release.

Reply:
> This can be fixed by changing Telegraf's logfmt dependency to ">=0.3.0, <0.5.0"

I can create a commit for this

> which is too strict and I expect any version >= 1.0.0 (the first release) would be fine.

@mstumpfx - since we have your eyes on this - what are your thoughts on making this change on the `golib` side?

Reply:
@amoghe I will talk to to a couple others about this today and figure out why we pin this revision, and if it can be updated. Will get back to you.

Reply:
Our protobuf version is tightly coupled to our client libraries so this may take a bit of work on our end, I'm still talking to a couple of the engineers who typically maintain the repo about it.

Reply:
Sorry for the delayed response - we're in the process of getting this prioritized on our end, and there's a few different teams involved. I don't have much finite information to share yet, but things are in motion.

Reply:
@mstumpfx - thanks for the update!